backend/connector-integration/src/connectors/phonepe/transformers.rs

```diff
@@ -0,0 +1,591 @@
+    pub intent_url: Option<String>, // For UPI Intent
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub qr_data: Option<String>, // For UPI QR
+}
+
+// PhonePe transaction details (optional in response)
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PhonePeTransactionDetails {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pay_response_code: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pay_response_code_description: Option<String>,
+}
+
+// PhonePe success response data
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PhonePeSuccessData {
+    pub merchant_id: String,
+    pub merchant_transaction_id: String,
+    pub instrument_response: PhonePeInstrumentResponse,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub response_code: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub response_code_description: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pay_response_code: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pay_response_code_description: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub transaction_details: Option<PhonePeTransactionDetails>,
+}
+
+// PhonePe success response structure
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PhonePeSuccessResponse {
+    pub success: bool,
+    pub code: String,
+    pub message: String,
+    pub data: PhonePeSuccessData,
+}
+
+// PhonePe error response structure
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PhonePeErrorResponse {
+    pub success: bool,
+    pub code: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub message: Option<String>,
+}
+
+// Combined PhonePe response type
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum PhonePePaymentResponse {
+    Success(PhonePeSuccessResponse),
+    Error(PhonePeErrorResponse),
+}
+
+
+
+
+impl
+    TryFrom<
+        
+            RouterDataV2<Authorize, PaymentFlowData, PaymentsAuthorizeData, PaymentsResponseData>,
+        
+    > for PhonePePaymentRequest
+{
+    type Error = error_stack::Report<hyperswitch_interfaces::errors::ConnectorError>;
+
+    fn try_from(
+        item: RouterDataV2<Authorize, PaymentFlowData, PaymentsAuthorizeData, PaymentsResponseData>,
+        
+    ) -> Result<Self, Self::Error> {
+        let router_data = &item;
+
+        // Extract auth credentials
+        let auth = PhonePeAuthType::try_from(&router_data.connector_auth_type)?;
+
+        // Prepare common request fields
+        let merchant_transaction_id = router_data.connector_request_reference_id.clone();
+        let amount = router_data.request.minor_amount.get_amount_as_i64();
```

Comment:
is this the expected way using amount framework? can you check with team and resolve this